File: Intermediate/Topic-003-Introduction-to-Arrays/Assigment/Q1-Good-Pair/good_pair.py
import sys
class Solution:
    # @param A : list of integers
    # @param B : integer
    # @return an integer
    def solve(self, A, B):
        if (len (A) <= 1):
            return 0

        for i in range (0, len(A)):
            for j in range (0, len(A)):
                if i != j:
                    if (B == A[i] + A[j]):
                        return 1
        
        return 0
    
## Runner code
if __name__ == "__main__":
    input_array = list(map(int, input("Enter the elements of the array separated by spaces: ").strip().split()))
    target_sum = int(input("Enter the target sum (B): "))
    solution = Solution()
    result = solution.solve(input_array, target_sum)
    if result == 1:
        print(f"A good pair exists in the array for the target sum {target_sum}.")
    else:
        print(f"No good pair exists in the array for the target sum {target_sum}.")

File: Intermediate/Topic-003-Introduction-to-Arrays/Assigment/Q1-Good-Pair/good_pair.c
 #include <stdio.h>
/**
 * @input A : Integer array
 * @input n1 : Integer array's ( A ) length
 * @input B : Integer
 * 
 * @Output Integer
 */
int solve(int* A, int n1, int B) {
    int idx, jdx;

    if (n1 <= 1) { 
        return 0;
    }
    for (idx = 0; idx < n1; idx++) {
        for (jdx= 0; jdx < n1; jdx++) {
            if (idx == jdx) {
                continue;
            } else {
                if ((A[idx] + A[jdx]) == B) {
                    return 1;
                }
                
            }
        }
    }
    return 0;
}



int main() {
    int A[] = {1, 2, 3, 4, 5};
    int n1 = 5;
    int B = 7;
    
    int result = solve(A, n1, B);
    
    if (result == 1) {
        printf("Good pair found\n");
    } else {
        printf("No good pair found\n");
    }
    
    return 0;
}